infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -161,6 +170,49 @@ public ServerContext(Map<String, Object> serverConfig) {
         }
     }
 
+    /**
+     * Cleanup the DataStore files with names that are prefixes of the specified
+     * fileName when so that the number of these files don't exceed the user-defined
+     * retention limit. Cleanup is always done on files with lower epochs.
+     */
+     void dataStoreFileCleanup(String fileName) {
+        String logDirPath = getServerConfig(String.class, "--log-path");
+        if (logDirPath == null) {
+            return;
+        }
+
+        File logDir = new File(logDirPath);
+        Set<String> prefixesToClean = getDsFilePrefixesForCleanup();
+        int numRetention = Integer.parseInt(getServerConfig(String.class, "--metadata-retention"));
+
+        prefixesToClean.stream()
+                .filter(fileName::startsWith)
+                .forEach(prefix -> {
+                    File[] foundFiles = logDir.listFiles((dir, name) -> name.startsWith(prefix));
+                    if (foundFiles == null || foundFiles.length <= numRetention) {
+                        log.debug("DataStore cleanup not started for prefix: {}.", prefix);
+                        return;
+                    }
+                    log.debug("Start cleaning up DataStore files with prefix: {}.", prefix);
+                    Arrays.stream(foundFiles)
+                            .sorted(Comparator.comparingInt(file -> {
+                                // Extract epoch number from file name and cast to int for comparision
+                                Matcher matcher = Pattern.compile("\\d+").matcher(file.getName());
+                                return matcher.find(prefix.length()) ? Integer.parseInt(matcher.group()) : 0;
+                            }))
+                            .limit(foundFiles.length - numRetention)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this subtraction operation to a "long". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2184)